### DISCUSSION TITLE
Thumbnail only shows red screen

### DISCUSSION THREAD

**Discussion by aweaver1fandm** [epoptes/epoptes#166] [Q&A]
Running epoptes on Ubuntu 20.04 on standalone (not LTSP) clients which we installed from the standard repos.  However, on the screen that shows all the clients, the thumbnails only show as the default red screen with std (like when the client is at a login screen)  It never shows a screen preview when someone logs in.  If we double click on a client the screen shows up fine and we can see that someone is logged in, what they're doing, etc, but the preview thumbnail still remains the same.  We have one client on Ubuntu 18.04 whose thumbnail does show up just fine.

Results from apt show epoptes-client:
Package: epoptes-client
Version: 20.01-1
Priority: extra
Section: universe/admin

Results from epoptes-client -c:
depth=0 C = AU, ST = Some-State, O = Internet Widgits Pty Ltd
verify error:num=18:self signed certificate
verify return:1
depth=0 C = AU, ST = Some-State, O = Internet Widgits Pty Ltd
verify return:1
DONE
Successfully fetched certificate from cs-40.fandm.edu:789

**Comment by alkisg** [comment]
On a client, login, open a terminal, and run `epoptes-client`.
It will replace the already running epoptes-client, so that you can see its output, error messages etc.
Check the server if you get thumbnails then.
If not, paste the terminal output, it will probably mention why it can't produce thumbnails.

Note that if you're using Wayland, thumbnails probably won't work there.

**Reply by AnaIESHuelin** [reply]
I have the same problem. The connection clients and server is working but i cannot see the secreens. When I run epoptes-client I got the following error:
![42217523-EAE3-4182-8682-E09A28AE4E6E](https://user-images.githubusercontent.com/114143990/191703574-3fb80fa1-a604-4494-80cd-c735abd17d3b.png)

**Reply by alkisg** [reply]
@AnaIESHuelin , if you're running wayland, screen sharing and thumbshots won't work there.
Use Xorg instead, you can select it in the login screen (GDM).

**Comment by aweaver1fandm** [comment]
So we purged the epoptes server and re-installed it.  As a result none of
our clients, including the one running Ubuntu 18.04, have thumbnails with a
preview.  We can still click on the computer to see what is going on

**Comment by alkisg** [comment]
Please do not reply inline to github emails, they show up terrible on github:
https://github.com/epoptes/epoptes/discussions/166#discussioncomment-3529599

For the test, where you ran `epoptes-client` on a client, you ran it as root.
Run it as a user instead, `/usr/sbin/epoptes-client`, not as root.
The `epoptes-client` process that sends the thumbnails is the user (session) process, not the root (system) process.

**Reply by aweaver1fandm** [reply]
When we run epoptes-client as the user then the thumbnail shows up immediately

**Comment by alkisg** [comment]
That means that your desktop environment is not launching /etc/xdg/autostart/epoptes-client at login.
Which desktop environment is that? Have you customized it?

**Reply by aweaver1fandm** [reply]
We just figured out that is the problem.  We had to remove the .desktop file (by your recommendation from ticket #160) because on Ubuntu 20.04 when the user tries to logoff, the logoff hangs for several minutes before actually logging out.  You wrote: 

"Unfortunately it appears that recent GNOME versions do not like that and think that the session hasn't terminated yet.

Patches welcome; personally I don't use GNOME anywhere as it has a whole lot of other problems, but I might try to have a look in time for Ubuntu 24.04! :D"

Unfortunately I haven't had enough time to look at a possible patch for this for GNOME.

**Comment by aweaver1fandm** [comment]
Still nothing.

~# epoptes-client
Setting WOL=g for enp0s31f6
Epoptes-client connecting to cs-40.fandm.edu:789...   ...done

When we click on the thumbnail, we can see the desktop no problem.  The
thumbnail preview just isn't showing up correctly.
We are running Ubuntu 20.04, gnome 3.36.8 with X11.  No Wayland

Like I said before, we have one system on Ubuntu 18.04 where the thumbnail
shows up just fine.

**Comment by alkisg** [comment]
If `epoptes-client` in a terminal doesn't show why it can't produce thumbshots, then you'll need to come to the chat room for live troubleshooting.
